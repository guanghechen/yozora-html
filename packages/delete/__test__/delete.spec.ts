import type { IDelete } from '@yozora/ast'
import { renderChildren } from '../../../jest.setup'
import renderDelete from '../src'

describe('snapshot', function () {
  it('basic', function () {
    const node = {
      type: 'delete',
      children: [
        {
          type: 'text',
          value: 'yozora is cool!',
        },
      ],
    }
    expect(renderDelete(node as IDelete, renderChildren)).toMatchSnapshot()
  })

  it('basic-2', function () {
    const node = {
      type: 'delete',
      children: [
        {
          type: 'emphasis',
          children: [
            {
              type: 'text',
              value: 'Hello',
            },
          ],
        },
        {
          type: 'text',
          value: ', ',
        },
        {
          type: 'strong',
          children: [
            {
              type: 'text',
              value: 'World',
            },
          ],
        },
      ],
    }
    expect(renderDelete(node as IDelete, renderChildren)).toMatchSnapshot()
  })
})
