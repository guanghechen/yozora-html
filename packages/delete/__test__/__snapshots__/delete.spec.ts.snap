// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot basic 1`] = `"<del class="yozora-delete"><span class="yozora-text">yozora is cool!</span></del>"`;

exports[`snapshot basic-2 1`] = `"<del class="yozora-delete"><em class="yozora-emphasis"><span class="yozora-text">Hello</span></em><span class="yozora-text">, </span><strong class="yozora-strong"><span class="yozora-text">World</span></strong></del>"`;
